### PR TITLE
[RF][PyROOT] Refactor: move RooWorkspace variable creation logic to C++ backend

### PR DESCRIPTION
**Pull request by yashnator**
# This Pull request:

Refactors the code and moves the logic of creating variables in RooWorkspace to C++-backend

## Changes:

Moves `make_json_for_variable` from `_rooworkspace.py` to `varJSONString` in `JSONIOUtils`

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

**Comment by guitargeek** [comment]
@phsft-bot build

File: roofit/hs3/src/JSONIOUtils.cxx [hunk 1]
```diff
@@ -1,5 +1,58 @@
...
+   if (jsonInfoDict.contains("max") && jsonInfoDict.contains("min")) {
+      maxVal = jsonInfoDict["max"];
+      minVal = jsonInfoDict["min"];
+      isRange = true;
+   }
+   if (!isConstant) {
+      val = (maxVal + minVal) / 2;
+   }
+   // Check if variable is at least a range or constant else throw error
+   if (isConstant || isRange) {
```
Comment (guitargeek) [inline roofit/hs3/src/JSONIOUtils.cxx:33]:
```suggestion
   if (!(isConstant || isRange)) {
      throw std::invalid_argument("Invalid Syntax: Please provide either 'value' or 'min' and 'max' or both");
   }
```
Then the rest of this function doesn't need to be indented and it's easier to read the control flow.

**Review by guitargeek — CHANGES_REQUESTED** [review]
Hi Yash, thanks for the PR! For a first round of comments, I only have a small request besides that you have to rebase your commit on top of ROOT `master` and push again to the PR branch, because there are merge conflicts now.

File: roofit/hs3/src/JSONIOUtils.cxx (hunk 1 above)
Comment (yashnator) [reply roofit/hs3/src/JSONIOUtils.cxx]: This went unnoticed; my bad. I've changed it now.

**Comment by guitargeek** [comment]
@phsft-bot build

**Comment by guitargeek** [comment]
@phsft-bot build

**Comment by guitargeek** [comment]
@phsft-bot build

**Review by guitargeek — APPROVED** [review]
LGTM, thanks a lot for moving all the logic to the C++ side and using RooFits JSON interface!

**Pull request merged**